server/server.go

```diff
@@ -279,20 +279,29 @@ type response struct {
 	id        uint32
 	op        protocol.Operation
 	reqOpcode protocol.Op
+	err       protocol.Error
 	// time just after the request was deserialized from the connection
 	reqBegin time.Time
 }
 
-func makeRespondResponse(req request, payload []byte) response {
-	return response{id: req.pkt.ID, op: protocol.MakeRespondOp(payload), reqOpcode: req.pkt.Opcode, reqBegin: req.reqBegin}
+func (s *Server) makeRespondResponse(req request, payload []byte, requestBegin time.Time) response {
+	s.stats.logRequestExecDuration(req.pkt.Opcode, requestBegin, protocol.ErrNone)
+	return response{id: req.pkt.ID, op: protocol.MakeRespondOp(payload), reqOpcode: req.pkt.Opcode, err: protocol.ErrNone, reqBegin: req.reqBegin}
 }
 
-func makePongResponse(req request, payload []byte) response {
-	return response{id: req.pkt.ID, op: protocol.MakePongOp(payload), reqOpcode: req.pkt.Opcode, reqBegin: req.reqBegin}
+func (s *Server) makePongResponse(req request, payload []byte, requestBegin time.Time) response {
+	s.stats.logRequestExecDuration(req.pkt.Opcode, requestBegin, protocol.ErrNone)
+	return response{id: req.pkt.ID, op: protocol.MakePongOp(payload), reqOpcode: req.pkt.Opcode, err: protocol.ErrNone, reqBegin: req.reqBegin}
 }
 
-func makeErrResponse(req request, err protocol.Error) response {
-	return response{id: req.pkt.ID, op: protocol.MakeErrorOp(err), reqOpcode: req.pkt.Opcode, reqBegin: req.reqBegin}
+func (s *Server) makeErrResponse(req request, err protocol.Error, requestBegin time.Time) response {
+	s.stats.logRequestExecDuration(req.pkt.Opcode, requestBegin, err)
+
+	// This metric can now be derived from the above, and can be removed once
+	// dashboards are updated to use it.
+	s.stats.logInvalid(req.pkt.Opcode)
```

Comment:
Why don't we just remove it now?

Reply:
I need to be able to continue using the existing metrics during the rollout of new ones, otherwise I could be blind to new regressions.